src/controllers/AuthLocalController.js

```diff
@@ -10,70 +10,139 @@ dotenv.config();
     }
   }
 
+  /**
+   * @description function to delete user
+   * @param {object} req user request object
+   * @param {object} res response object from server
+   * @returns {object} return true if user deleted or false when user not deleted
+   */
+  static async deactivateAccount(req, res) {
+    const { id } = req.params;
+    const deactivateAccount = await User.update({ isActive: false }, { id });
+    return deactivateAccount
+      ? res.status(status.OK).json({ message: 'User account deleted successfully' })
+      : res.status(status.UNAUTHORIZED).json({ errors: 'Unauthorized access' });
+  }
+
+  /**
+   * @description methode to find one user
+   * @param {object} req user request object
+   * @param {object} res response object from server
+   * @returns {object} return all users
+   */
+  static async getOne(req, res) {
+    const id = Number.parseInt(req.params.id, 10);
+    const fetchUser = await User.findOne({ id: Number.isNaN(id) ? 0 : id });
+    delete fetchUser.password;
+    return Object.keys(fetchUser).length
+      ? res.status(status.OK).json({ fetchUser })
+      : res
+        .status(status.NOT_FOUND)
+        .json({ errors: { user: `sorry, user with id "${req.params.id}" not found!!` } });
+  }
+
+  /**
+   * @description function for admin to create user
+   * @param {object} req user request object
+   * @param {object} res response object from server
+   * @returns {object} return true if user created or flase when was not
+   */
+  static async create(req, res) {
+    const {
+      password, email, firstName, lastName
+    } = req.body;
+    req.body.password = helper.password.hash(req.body.password);
+    const newUser = await User.create(req.body);
+    if (newUser.errors) {
+      const errors = helper.checkCreateUpdateUserErrors(newUser.errors);
+      const { code } = errors;
+      delete errors.code;
+      return res.status(code).json(errors);
+    }
+    if (newUser) {
+      await helper.sendMail(email, 'resetPassword', { firstName, lastName, password });
+      return res.status(status.CREATED).json({
+        message: `activetion message sent to ${req.body.email}`
+      });
+    }
+  }
+
+  /**
+   * @description function to activate user account
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} it return true if account activeted otherwise it return false
+   */
+  static async activate(req, res) {
+    const { user } = req;
+    const activated = await User.update({ isActive: true }, { email: user.email });
+    await helper.sendMail(activated.email, 'accountActivatedMsg', {
+      firstName: activated.firstName
+    });
+    return res.status(status.OK).json({
+      message: 'Account successfully activated'
+    });
+  }
+
+  /**
+   * @description function to activate user account
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} it return true if account verified otherwise it return false
+   */
+  static async reset(req, res) {
```

Comment:
the token should be sent to user email

Reply:
yes the token sent to the user email however, I added this function to return it on the body I can remove it

Reply:
shouldn't be returned. Just send email.